### PR TITLE
TaskState refactor

### PR DESCRIPTION
**Pull request by alexandraorth**

### CODE REVIEW

File: grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/StandaloneTaskManager.java [hunk 1]
```diff
@@ -169,22 +168,20 @@ public TaskStateStorage storage() {
...
-                stateStorage.updateState(state
-                        .status(FAILED)
-                        .statusChangedBy(EXCEPTION_CATCHER_NAME)
-                        .exception(getFullStackTrace(t)));
+            catch (Throwable throwable) {
+                LOG.error(getFullStackTrace(throwable));
+                state.markFailed(throwable);
             }
+            stateStorage.updateState(state);
+            stateUpdateLock.unlock();
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/StandaloneTaskManager.java]: Might need to go in a `finally`.

File: grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/ZookeeperConnection.java [hunk 2]
```diff
@@ -84,7 +83,7 @@ public CuratorFramework connection(){
         return zookeeperConnection;
     }
 
-    public InterProcessMutex mutex(TaskId id){
+    public InterProcessMutex mutex(String id){
```
Comment (aelred) [inline grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/ZookeeperConnection.java]: :(

File: grakn-graql/src/main/java/ai/grakn/graql/internal/template/TemplateVisitor.java [hunk 3]
```diff
@@ -346,7 +346,7 @@ public String visitReplaceStatement(GraqlTemplateParser.ReplaceStatementContext
         if(value == ObjectUtils.NULL){
             throw new GraqlTemplateParsingException("Key [" + ctx.getText() + "] not present in data: " + originalContext);
         }
-
+        ;
```
Comment (aelred) [inline grakn-graql/src/main/java/ai/grakn/graql/internal/template/TemplateVisitor.java]: good stuff

File: grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/ZookeeperConnection.java (hunk 2 above)
Comment (alexandraorth) [inline grakn-engine/src/main/java/ai/grakn/engine/tasks/manager/ZookeeperConnection.java]: You can now get a mutex on the engine ID or the task ID. This is why I was asking about that ID interface earlier ;) Related: in `TaskId` we use `getValue` and in `EngineID` we use `value()`

**Pull request merged**